### PR TITLE
[enh] Add the initial support of contributed representations

### PR DESCRIPTION
**Pull request by gcoutable**
### Type of this PR 

- [x] New feature or improvement

### What does this PR do?

It eases the contribution of new kinds of representation in the front-end

### CODE REVIEW

File: frontend/src/properties/PropertiesWebSocketContainerMachine.ts [hunk 1]
```diff
@@ -217,6 +217,7 @@ export const propertiesWebSocketContainerMachine = Machine<
           selection.kind === 'Unknown' ||
           selection.kind === 'Diagram' ||
           selection.kind === 'Form' ||
+          selection.kind === 'ForceDirectedTree' || // Should be removed
```
Comment (sbegaudeau) [inline frontend/src/properties/PropertiesWebSocketContainerMachine.ts:220]: Can't you use RepresentationComponentRegistry#isRepresentation here?

Comment (gcoutable) [reply frontend/src/properties/PropertiesWebSocketContainerMachine.ts:220]: I don't think I can. The registry is not exported and I don't think I have access to `useContext` here.

Comment (sbegaudeau) [reply frontend/src/properties/PropertiesWebSocketContainerMachine.ts:220]: You can use `useContext` in `PropertiesWebSocketContainer` and indicate if the selection is a representation in the SwitchSelectionEvent.

Comment (gcoutable) [reply frontend/src/properties/PropertiesWebSocketContainerMachine.ts]: Done

**Review by pcdavid — APPROVED** [review]

**Pull request merged**